blaze-core/src/main/scala/org/http4s/blazecore/util/CachingChunkWriter.scala

```diff
@@ -42,37 +42,44 @@ private[http4s] class CachingChunkWriter[F[_]](
   import ChunkWriter._
 
   private[this] var pendingHeaders: StringWriter = _
-  private[this] var bodyBuffer: Buffer[Chunk[Byte]] = Buffer()
+  private[this] val bodyBuffer: Buffer[Chunk[Byte]] = Buffer()
   private[this] var size: Int = 0
 
   override def writeHeaders(headerWriter: StringWriter): Future[Unit] = {
     pendingHeaders = headerWriter
     FutureUnit
   }
 
-  private def addChunk(chunk: Chunk[Byte]): Unit = {
-    bodyBuffer += chunk
-    size += chunk.size
-  }
-
-  private def clear(): Unit = {
+  private def addChunk(chunk: Chunk[Byte]): Unit =
+    if (chunk.nonEmpty) {
+      bodyBuffer += chunk
+      size += chunk.size
+    }
+
+  private def toChunkAndClear: Chunk[Byte] = {
```

Comment:
Even if `toChunk` and `clear` are coupled in use, it's better to have these atomic ("a function do one thing and is good at it"). But feel free to skip this grouchy nitpick.

Reply:
Sorry, I'm not quite following.  Are you proposing splitting this into two?

Reply:
Yep, just as it was previously. But, I'm not too sure about this take. So nevermind.